crates/runtime/src/runtime/default.rs

```diff
@@ -599,15 +601,24 @@ where
         let template_address = base.account.template_addr();
 
         let template = self.env.template(template_address, None).unwrap();
-
         let template_code_section = template.sections().get(SectionKind::Code).as_code();
-        let gas_mode = template_code_section.gas_mode();
         let template_code = template_code_section.code();
+        let gas_mode = template_code_section.gas_mode();
         let program = Program::new(template_code, false).unwrap();
-        let func_price = {
+
+        // We're using a naive memoization mechanism: we only ever add, never
+        // remove. This means there's no cache invalidation at all. We can
+        // easily afford to do this because the number of templates that exist
+        // at genesis is fixed and won't grow.
```

Comment:
Please also comment that once we'll have more `PriceResolver` used the caching keys will consist not only of a `TemplateAddr` but of `PriceResolver Id` (this `id` might be `integer/string` etc).

That way, each `Template` will hold multiple cached `FuncPrice`.
The parameters for resolving a `Price Resolver` will be determined by what we'll decide in the future.
It could be a `LayerId` or the `svm_version` the `Template` has been compiled against (or both `LayerId` and `svm_version`).

In any case, mapping using `PriceResolver Id` will be ignorant to the parameters given to the `PriceResolverRegistry`.

Reply:
See the documentation for method `with_template_prices`